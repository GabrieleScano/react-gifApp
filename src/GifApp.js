import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div>
      <h2>Gif App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ul>
        {categories.map((category) => {
          return <GifGrid category={category} key={category} />;
        })}
      </ul>

      <hr />
    </div>
  );
};

export default GifApp
