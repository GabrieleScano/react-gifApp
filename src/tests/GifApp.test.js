import React from 'react'
import  GifApp  from '../GifApp'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

describe('Test <GifApp /> component', () => {

    test('should show the component', () => {
        const wrapper = shallow(<GifApp />)
        expect(wrapper).toMatchSnapshot()
      });

      test('should show a list of categories', () => {
          const categories = ['one', 'two']
          const wrapper = shallow(<GifApp defaultCategories={categories}/>)
          expect(wrapper).toMatchSnapshot()
          expect(wrapper.find('GifGrid').length).toBe(categories.length)
          
      })
      
    
})

