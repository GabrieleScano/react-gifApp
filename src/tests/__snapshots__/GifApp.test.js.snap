// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifApp /> component should show a list of categories 1`] = `
<div>
  <h2>
    Gif App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="one"
      key="one"
    />
    <GifGrid
      category="two"
      key="two"
    />
  </ul>
  <hr />
</div>
`;

exports[`Test <GifApp /> component should show the component 1`] = `
<div>
  <h2>
    Gif App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
  <hr />
</div>
`;
